### PR TITLE
api: update time parameter to datetime per WFS 3 change

### PR DESCRIPTION
**Pull request by philvarner**

### CODE REVIEW

File: api-spec/STAC-extensions.yaml [hunk 1]
```diff
@@ -761,35 +769,44 @@ components:
...
         intersects:
           $ref: 'https://geojson.org/schema/Geometry.json'
-    time:
+    datetimeFilter:
+      description: An object representing a datetime-based filter.
+      type: object
+      properties:
+        datetime:
+          $ref: '#/components/schemas/datetimeOrDatetimeInterval'
+    datetimeOrDatetimeInterval:
```
Comment (philvarner) [inline api-spec/STAC-extensions.yaml]: we had two uses of this type, but one should only have allowed an RFC 3339 datetime, while the other allowed that or the `datetime/datetime` interval syntax, so split them into two definitions.

File: api-spec/STAC-extensions.yaml [hunk 2]
```diff
@@ -761,35 +769,44 @@ components:
       properties:
         bbox:
           $ref: '#/components/schemas/bbox'
-    timeFilter:
```
Comment (philvarner) [inline api-spec/STAC-extensions.yaml]: moved this down so bbox and intersects would be next to one another, and the 3 datetime-related ones would be also.

File: api-spec/STAC-extensions.yaml (hunk 1 above)
Comment (m-mohr) [reply api-spec/STAC-extensions.yaml]: This is how WFS defines it. https://raw.githubusercontent.com/opengeospatial/WFS_FES/master/core/openapi/parameters/datetime.yaml

Should we follow them?

**Comment by matthewhanson** [comment]
@philvarner Does WFS3 allow for specifying incomplete datetimes?  That is, can I specify just the date?  This is of course the main use case, by a wide margin, and I'd like to not have to specify a time.

Being able to specify a single datetime (and not an interval) isn't that useful unless I can specify only the date.  This means that it really is an interval of a single day, but I think the shortcut is valuable.

**Comment by m-mohr** [comment]
> @philvarner Does WFS3 allow for specifying incomplete datetimes?

No.


> This is of course the main use case, by a wide margin, and I'd like to not have to specify a time.

Clients could extend `2019-01-01` to `2019-01-01T00:00:00Z/2019-01-02T00:00:00Z` I guess?!

File: api-spec/STAC-extensions.yaml (hunk 1 above)
Comment (philvarner) [reply api-spec/STAC-extensions.yaml]: > Should we follow them?

I'll update this wrt their definition

**Comment by m-mohr** [comment]
After aligning with WFS3 draft2 in #525, I would propose the following changes:
* Follow WFS3 strictly regarding datetime
* Add another query parameter `date`, which allows exactly the same, but for dates only. This would cover what @matthewhanson critizied in #521. Examples
    * `?date=2017-01-01` to cover the whole 1st of Jan 2017
    * `?date=2017-01-01/2017-03-31` to cover Jan/Feb/Mar

**Comment by philvarner** [comment]
>After aligning with WFS3 draft2 in #525, I would propose the following changes:
> Follow WFS3 strictly regarding datetime (I could include this in #525)

Yes.

> Add another query parameter date in this PR, which allows exactly the same, but for dates only. This > would cover what @matthewhanson critizied in #521. Examples
> ?date=2017-01-01 to cover the whole 1st of Jan 2017
> ?date=2017-01-01/2017-03-31 to cover Jan/Feb/Mar

I don't think this is necessary, as one can do `datetime=2017-01-01T00:00:00Z/2017-01-31T23:59:59.999Z`  (thought a separate clarification is whether the semantics of WFS 3 is that the time is >= to the start and <= end, as it's not explicitly clear in that spec.  I've usually implemented the semantics that it's >= start and < end to facilitate month query)

**Comment by m-mohr** [comment]
> I don't think this is necessary, as one can do `datetime=2017-01-01T00:00:00Z/2017-01-31T23:59:59.999Z` (thought a separate clarification is whether the semantics of WFS 3 is that the time is >= to the start and <= end, as it's not explicitly clear in that spec. I've usually implemented the semantics that it's >= start and < end to facilitate month query)

I've asked Clemens (WFS editor) some days ago and he said they are following ISO8601 and it is >= and <=. I don't like that and prefer >= and < as you do, but I think that has been discussed already in the WFS group and we can't change it. Therefore, we probably just follow their decision.

**Comment by m-mohr** [comment]
I ported the changes proposed here to #529 . Discussions regarding a date query parameter can be continued in #536.

**Pull request closed without merging**